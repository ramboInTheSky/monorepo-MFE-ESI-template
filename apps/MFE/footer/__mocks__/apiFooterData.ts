const apiFooterData = {
    id: "ab2f7c47-c12e-4340-b782-033a1cb2b2db",
    name: "homePage",
    realm: "amido",
    territory: "GB",
    language: "en",
    regions: [
        {
            type: "SocialMedia",
            title: "Our Social Networks",
            accessibilityTitle: "Our Social Networks",
            subRegions: [
                {
                    title: "",
                    accessibilityTitle: "",
                    elements: [
                        {
                            type: "Link",
                            url: "https://www.facebook.com/amidoofficial",
                            openInNewWindow: true,
                            name: "Facebook",
                            icon: "/icons/footer/amido/socialmedia/icon-social-facebook.svg",
                            text: "",
                            accessibilityText: "Amido Official Facebook page (opens in a new window)",
                            tooltip: "",
                            accessibilityTooltip: "",
                            description: "",
                            accessibilityDescription: "",
                        },
                        {
                            type: "Link",
                            url: "https://www.twitter.com/amidoofficial",
                            openInNewWindow: true,
                            name: "Twitter",
                            icon: "/icons/footer/amido/socialmedia/icon-social-twitter.svg",
                            text: "",
                            accessibilityText: "Amido Official Twitter page (opens in a new window)",
                            tooltip: "",
                            accessibilityTooltip: "",
                            description: "",
                            accessibilityDescription: "",
                        },
                        {
                            type: "Link",
                            url: "https://instagram.com/amidoofficial",
                            openInNewWindow: true,
                            name: "Instagram",
                            icon: "/icons/footer/amido/socialmedia/icon-social-instagram.svg",
                            text: "",
                            accessibilityText: "Amido Official Instagram page (opens in a new window)",
                            tooltip: "",
                            accessibilityTooltip: "",
                            description: "",
                            accessibilityDescription: "",
                        },
                        {
                            type: "Link",
                            url: "pinterest.com/amidoofficial",
                            openInNewWindow: true,
                            name: "Pinterest",
                            icon: "/icons/footer/amido/socialmedia/icon-social-pinterest.svg",
                            text: "Pinterest",
                            accessibilityText: "pinterest",
                            tooltip: "pinterest.com/amidoofficial",
                            accessibilityTooltip: "pinterest.com/amidoofficial",
                            description: "",
                            accessibilityDescription: "",
                        },
                        {
                            type: "Link",
                            url: "https://www.youtube.com/user/amidoshopping",
                            openInNewWindow: true,
                            name: "Youtube",
                            icon: "/icons/footer/amido/socialmedia/icon-social-youtube.svg",
                            text: "",
                            accessibilityText: "Amido Shopping Youtube page (opens in a new window)",
                            tooltip: "",
                            accessibilityTooltip: "",
                            description: "",
                            accessibilityDescription: "",
                        },
                    ],
                },
            ],
        },
        {
            type: "QuickLinks",
            title: "",
            accessibilityTitle: "",
            subRegions: [
                {
                    title: "",
                    accessibilityTitle: "",
                    elements: [
                        {
                            type: "MyAccountLoggedIn",
                            url: "https://www.amido.com/secure/accounts/transfer",
                            openInNewWindow: false,
                            name: "MyAccount",
                            icon: "/icons/footer/amido/quicklinks/myaccount.svg",
                            text: "<username>",
                            accessibilityText: "My Account",
                            tooltip: "My Account",
                            accessibilityTooltip: "My Account",
                            description: "Sign-in to your account",
                            accessibilityDescription: "Sign-in to your account",
                        },
                        {
                            type: "MyAccountLoggedOut",
                            url: "https://www.amido.com/secure/accounts/transfer",
                            openInNewWindow: false,
                            name: "MyAccount",
                            icon: "/icons/footer/amido/quicklinks/myaccount.svg",
                            text: "My Account",
                            accessibilityText: "My Account",
                            tooltip: "My Account",
                            accessibilityTooltip: "My Account",
                            description: "View your Amido account",
                            accessibilityDescription: "View your Amido account",
                        },
                        {
                            type: "Link",
                            url: "https://www.amido.com/storelocator",
                            openInNewWindow: false,
                            name: "StoreLocator",
                            icon: "/icons/footer/amido/quicklinks/storelocator.svg",
                            text: "Store Locator",
                            accessibilityText: "Store Locator",
                            tooltip: "Store Locator",
                            accessibilityTooltip: "Store Locator",
                            description: "For your nearest store",
                            accessibilityDescription: "For your nearest store",
                        },
                        {
                            type: "Link",
                            url: "https://www.amido.com/startachat",
                            openInNewWindow: false,
                            name: "StartAChat",
                            icon: "/icons/footer/amido/quicklinks/startachat.svg",
                            text: "Start A Chat",
                            accessibilityText: "Start A Chat",
                            tooltip: "Start A Chat",
                            accessibilityTooltip: "Start A Chat",
                            description: "For general enquiries",
                            accessibilityDescription: "For general enquiries",
                        },
                        {
                            type: "LanguageSelector",
                            url: "",
                            openInNewWindow: false,
                            name: "LanguageSelector",
                            icon: "/icons/footer/amido/quicklinks/languageselector.svg",
                            text: "Select Language",
                            accessibilityText: "Select Language",
                            tooltip: "Select Language",
                            accessibilityTooltip: "Select Language",
                            description: "",
                            accessibilityDescription: "",
                        },
                    ],
                },
                {
                    title: "Logout",
                    accessibilityTitle: "",
                    elements: [
                        {
                            type: "Logout",
                            url: "https://www.amido.com/logout",
                            openInNewWindow: false,
                            name: "Logout",
                            icon: "account_circle",
                            text: "Not <username>?",
                            accessibilityText: "Not <username>?",
                            tooltip: "Logout",
                            accessibilityTooltip: "Logout",
                            description: "Logout of account",
                            accessibilityDescription: "Logout of account",
                        },
                        {
                            type: "LogoutButton",
                            url: "https://www.amido.com/logout",
                            openInNewWindow: false,
                            name: "LogoutButton",
                            icon: "account_circle",
                            text: "Sign Out",
                            accessibilityText: "Sign Out",
                            tooltip: "Logout",
                            accessibilityTooltip: "Logout",
                            description: "Logout of your account",
                            accessibilityDescription: "Logout of your account",
                        },
                    ],
                },
            ],
        },
        {
            type: "MainLinks",
            title: "",
            accessibilityTitle: "",
            subRegions: [
                {
                    title: "Help",
                    accessibilityTitle: "Help",
                    elements: [
                        {
                            type: "Link",
                            url: "https://www.amido.com/help",
                            openInNewWindow: true,
                            name: "Help",
                            icon: "",
                            text: "View Help Topics",
                            accessibilityText: "View Help Topics",
                            tooltip: "View Help Topics",
                            accessibilityTooltip: "View Help Topics",
                            description: "",
                            accessibilityDescription: "",
                        },
                        {
                            type: "Link",
                            url: "https://www.accessable.co.uk/organisations/amido",
                            openInNewWindow: true,
                            name: "Store Accessibility",
                            icon: "",
                            text: "Accessibility In Our Stores",
                            accessibilityText: "Accessibility In Our Stores",
                            tooltip: "Accessibility In Our Stores",
                            accessibilityTooltip: "Accessibility In Our Stores",
                            description: "",
                            accessibilityDescription: "",
                        },
                        {
                            type: "Link",
                            url: "xxxxx",
                            openInNewWindow: true,
                            name: "Sitemap",
                            icon: "",
                            text: "Site Map",
                            accessibilityText: "Site Map",
                            tooltip: "Site Map",
                            accessibilityTooltip: "Site Map",
                            description: "",
                            accessibilityDescription: "",
                        },
                        {
                            type: "Link",
                            url: "tel: 0333 777 8000",
                            openInNewWindow: true,
                            name: "Customer Services",
                            icon: "",
                            text: "Customer Services- 0333 777 8000",
                            accessibilityText: "Customer Services telephone number",
                            tooltip: "Customer Services",
                            accessibilityTooltip: "Customer Services",
                            description: "Check your service provider for charges",
                            accessibilityDescription: "Check your service provider for charges",
                        },
                        {
                            type: "Link",
                            url: "https://www.amido.com/help/Faq/32475",
                            openInNewWindow: true,
                            name: "Contact Us",
                            icon: "",
                            text: "Contact Us",
                            accessibilityText: "Contact Us",
                            tooltip: "Contact Us",
                            accessibilityTooltip: "Contact Us",
                            description: "",
                            accessibilityDescription: "",
                        },
                    ],
                },
                {
                    title: "Privacy & Legal",
                    accessibilityTitle: "Privacy & Legal",
                    elements: [
                        {
                            type: "Link",
                            url: "/privacy",
                            openInNewWindow: false,
                            name: "Cookies & Privacy Policy",
                            icon: "",
                            text: "Cookies & Privacy Policy",
                            accessibilityText: "Cookies & Privacy Policy",
                            tooltip: "Cookies & Privacy Policy",
                            accessibilityTooltip: "Cookies & Privacy Policy",
                            description: "",
                            accessibilityDescription: "",
                        },
                        {
                            type: "Link",
                            url: "/terms",
                            openInNewWindow: false,
                            name: "Terms & Conditions",
                            icon: "",
                            text: "Terms & Conditions",
                            accessibilityText: "Terms & Conditions",
                            tooltip: "Terms & Conditions",
                            accessibilityTooltip: "Terms & Conditions",
                            description: "",
                            accessibilityDescription: "",
                        },
                        {
                            type: "Link",
                            url: "http://www.amido.com",
                            openInNewWindow: true,
                            name: "The Company",
                            icon: "",
                            text: "The Company",
                            accessibilityText: "The Company",
                            tooltip: "The Company",
                            accessibilityTooltip: "The Company",
                            description: "",
                            accessibilityDescription: "",
                        },
                        {
                            type: "Link",
                            url: "https://www.amido.com/media/media-contacts",
                            openInNewWindow: true,
                            name: "Media & Press",
                            icon: "",
                            text: "Media & Press",
                            accessibilityText: "Media & Press",
                            tooltip: "Media & Press",
                            accessibilityTooltip: "Media & Press",
                            description: "",
                            accessibilityDescription: "",
                        },
                        {
                            type: "Link",
                            url: "https://careers.amido.com",
                            openInNewWindow: true,
                            name: "Amido Careers",
                            icon: "",
                            text: "Careers @ Amido",
                            accessibilityText: "Careers at Amido",
                            tooltip: "Amido Careers",
                            accessibilityTooltip: "Amido Careers",
                            description: "",
                            accessibilityDescription: "",
                        },
                    ],
                },
                {
                    title: "Other Services",
                    accessibilityTitle: "Other Services",
                    elements: [
                        {
                            type: "Link",
                            url: "/amidopay",
                            openInNewWindow: false,
                            name: "AmidoPay",
                            icon: "",
                            text: "AmidoPay Credit Account",
                            accessibilityText: "AmidoPay Credit Account",
                            tooltip: "AmidoPay Credit Account",
                            accessibilityTooltip: "AmidoPay Credit Account",
                            description: "",
                            accessibilityDescription: "",
                        },
                        {
                            type: "Link",
                            url: "/unlimited",
                            openInNewWindow: false,
                            name: "Amido Unlimited",
                            icon: "",
                            text: "Amido Unlimited",
                            accessibilityText: "Amido Unlimited",
                            tooltip: "Amido Unlimited",
                            accessibilityTooltip: "Amido Unlimited",
                            description: "",
                            accessibilityDescription: "",
                        },
                        {
                            type: "Link",
                            url: "https://flowers.amido.com",
                            openInNewWindow: true,
                            name: "Flowers",
                            icon: "",
                            text: "Flowers, Plants & Wine",
                            accessibilityText: "Flowers, Plants & Wine",
                            tooltip: "Flowers, Plants & Wine",
                            accessibilityTooltip: "Flowers, Plants & Wine",
                            description: "",
                            accessibilityDescription: "",
                        },
                        {
                            type: "Link",
                            url: "https://flowers.amido.com/gift-cards",
                            openInNewWindow: true,
                            name: "Gift Cards",
                            icon: "",
                            text: "Gift Cards",
                            accessibilityText: "Gift Cards",
                            tooltip: "Gift Cards",
                            accessibilityTooltip: "Gift Cards",
                            description: "",
                            accessibilityDescription: "",
                        },
                        {
                            type: "Link",
                            url: "https://www.amido.com/evouchers/",
                            openInNewWindow: true,
                            name: "Evouchers",
                            icon: "",
                            text: "eGift Cards",
                            accessibilityText: "ee Gift Cards",
                            tooltip: "eGift Cards",
                            accessibilityTooltip: "ee Gift Cards",
                            description: "",
                            accessibilityDescription: "",
                        },
                        {
                            type: "Link",
                            url: "/shop/department-homeware-category-giftexperiences-0",
                            openInNewWindow: true,
                            name: "Gift Experiences",
                            icon: "",
                            text: "Gift Experiences",
                            accessibilityText: "Gift Experiences",
                            tooltip: "Gift Experiences",
                            accessibilityTooltip: "Gift Experiences",
                            description: "",
                            accessibilityDescription: "",
                        },
                    ],
                },
                {
                    title: "Shop By Country",
                    accessibilityTitle: "Shop By Country",
                    elements: [
                        {
                            type: "CountryLink",
                            url: "",
                            openInNewWindow: false,
                            name: "CountryLink",
                            icon: "",
                            text: "Saudi Arabia",
                            accessibilityText: "Saudi Arabia",
                            tooltip: "",
                            accessibilityTooltip: "",
                            description: "",
                            accessibilityDescription: "",
                        },
                        {
                            type: "CountryLanguage",
                            url: "www.amido.com/sa/ar",
                            openInNewWindow: false,
                            name: "CountryLanguage",
                            icon: "",
                            text: "AR",
                            accessibilityText: "Change to Saudi Arabia in Arabic",
                            tooltip: "",
                            accessibilityTooltip: "",
                            description: "",
                            accessibilityDescription: "",
                        },
                        {
                            type: "CountryLanguage",
                            url: "www.amido.com/sa/en",
                            openInNewWindow: false,
                            name: "CountryLanguage",
                            icon: "",
                            text: "EN",
                            accessibilityText: "Change to Saudi Arabia in English",
                            tooltip: "",
                            accessibilityTooltip: "",
                            description: "",
                            accessibilityDescription: "",
                        },
                        {
                            type: "CountryLink",
                            url: "",
                            openInNewWindow: false,
                            name: "CountryLink",
                            icon: "",
                            text: "United Arab Emirates",
                            accessibilityText: "United Arab Emirates",
                            tooltip: "",
                            accessibilityTooltip: "",
                            description: "",
                            accessibilityDescription: "",
                        },
                        {
                            type: "CountryLanguage",
                            url: "www.amido.com/ae/ar",
                            openInNewWindow: false,
                            name: "CountryLanguage",
                            icon: "",
                            text: "AR",
                            accessibilityText: "Change to United Arab Emirates in Arabic",
                            tooltip: "",
                            accessibilityTooltip: "",
                            description: "",
                            accessibilityDescription: "",
                        },
                    ],
                },
            ],
        },
        {
            type: "Copyright",
            title: "",
            accessibilityTitle: "",
            subRegions: [
                {
                    title: "",
                    accessibilityTitle: "",
                    elements: [
                        {
                            type: "MobileToggle",
                            url: "/changedevice/mobile",
                            openInNewWindow: false,
                            name: "MobileToggle",
                            icon: "",
                            text: "View Mobile Site",
                            accessibilityText: "View Mobile Site",
                            tooltip: "View Mobile Site",
                            accessibilityTooltip: "View Mobile Site",
                            description: "View Mobile Site",
                            accessibilityDescription: "View Mobile Site",
                        },
                        {
                            type: "DesktopToggle",
                            url: "/changedevice/desktop",
                            openInNewWindow: false,
                            name: "DesktopToggle",
                            icon: "",
                            text: "View Desktop Site",
                            accessibilityText: "View Desktop Site",
                            tooltip: "View Desktop Site",
                            accessibilityTooltip: "View Desktop Site",
                            description: "View Desktop Site",
                            accessibilityDescription: "View Desktop Site",
                        },
                        {
                            type: "Copyright",
                            url: "",
                            openInNewWindow: false,
                            name: "Copyright",
                            icon: "",
                            text: "© 2020 Amido RETAIL LTD. ALL RIGHTS RESERVED.",
                            accessibilityText: "© 2020 Amido RETAIL LTD. ALL RIGHTS RESERVED.",
                            tooltip: "Copyright",
                            accessibilityTooltip: "Copyright",
                            description: "Copyright",
                            accessibilityDescription: "Copyright",
                        },
                    ],
                },
            ],
        },
    ],
}

export default apiFooterData
