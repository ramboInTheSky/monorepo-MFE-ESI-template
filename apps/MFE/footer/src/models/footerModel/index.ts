export class FooterModel {
    public id = ""
    public name = ""
    public realm = ""
    public territory = ""
    public language = ""
    public regions: RegionModel[] = []
}

export class RegionModel {
    public type = ""
    public title = ""
    public accessibilityTitle = ""
    public subRegions: SubRegionModel[] = []
}

export class SubRegionModel {
    public title = ""
    public accessibilityTitle = ""
    public elements: SubRegionElementModel[] = []
}

export class SubRegionElementModel {
    public url = ""
    public openInNewWindow = false
    public name = ""
    public type = ""
    public icon = ""
    public text = ""
    public accessibilityText = ""
    public tooltip = ""
    public accessibilityTooltip = ""
    public description = ""
    public accessibilityDescription = ""
}
