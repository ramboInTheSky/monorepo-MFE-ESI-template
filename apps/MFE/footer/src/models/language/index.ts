export interface Language {
    id: string
    displayText: string
    name: string
    url: string
}

export interface ReturnLanguage {
    Language: Language
    isActiveLanguage: boolean
}

export class LanguageOptions {
    Languages: Array<ReturnLanguage> = []
}

export enum Direction {
    RTL = 'rtl',
    LTR = 'ltr'
}