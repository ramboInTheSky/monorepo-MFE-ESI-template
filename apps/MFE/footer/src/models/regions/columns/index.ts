type ColumnRange = 1 | 2 | 3 | 4 | 6 | 12

export interface DefaultColumnRanges {
    xs: ColumnRange
    sm: ColumnRange
    lg: ColumnRange
}

export const DefaultColumns: DefaultColumnRanges = {
    xs: 12,
    sm: 6,
    lg: 3,
}
