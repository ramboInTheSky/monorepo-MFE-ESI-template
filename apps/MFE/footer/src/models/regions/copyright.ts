export enum SupportedCopyrightTypes {
    Copyright = "Copyright",
    DesktopToggle = "DesktopToggle",
    MobileToggle = "MobileToggle",
}

export enum DeviceType {
    Desktop = "Desktop",
    Mobile = "Mobile",
}

export const deviceSwitcherUtils = {
    regex: "=([^;]+)",
    cookieKey: "AmidoDeviceType",
}
