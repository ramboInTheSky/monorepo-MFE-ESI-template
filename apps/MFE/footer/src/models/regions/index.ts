export * from "./copyright"
export * from "./columns"

export enum SupportedRegionTypes {
    SocialMedia = "SocialMedia",
    QuickLinks = "QuickLinks",
    MainLinks = "MainLinks",
    Copyright = "Copyright",
}
