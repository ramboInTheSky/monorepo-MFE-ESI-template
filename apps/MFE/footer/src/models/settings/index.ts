export enum SettingsSdkKeys {
    direction = "footer.frontend.direction",
    realm = "footer.frontend.realmName",
    territory = "footer.frontend.territoryName",
    language = "footer.frontend.languageName",
    appCache = "footer.frontend.appCacheTTL",
    languageId = "footer.alternativeLanguages",
    correlationHeaderName = "x-monorepo-correlation-id",
    sessionHeaderName = "x-monorepo-session-id",
    cacheControl = "footer.frontend.cache-control.max-age",
    cacheControlDefaultFooter = "footer.frontend.cache-control.default-footer.max-age",
    themeVersion = "footer.frontend.themeVersion",
    textData = "footer.frontend.textData",
    variant = "footer.frontend.variant",
}

export interface SettingsModel {
    realm: string
    alignment: string
    territory: string
    language: string
}
