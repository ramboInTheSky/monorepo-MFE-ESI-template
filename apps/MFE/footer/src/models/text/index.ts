export interface TextModel {
    defaultCopyrightText: string
  }