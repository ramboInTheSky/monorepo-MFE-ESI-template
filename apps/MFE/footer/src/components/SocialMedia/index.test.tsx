import React from "react"
import {render} from "@testing-library/react"
import {SCThemeProvider} from "@monorepo/theme-provider"
import {mockTheme} from "@monorepo/themes"
import apiFooterData from "../../../__mocks__/apiFooterData"
import {SupportedRegionTypes} from "../../models/regions"
import {SocialMedia} from "."

const socialMediaData = apiFooterData.regions.find(region => region.type === SupportedRegionTypes.SocialMedia)

const props = {
    region: socialMediaData,
    variant: "default",
    realm: "amido",
}

describe("Components - SocialMedia: ", () => {
    it("should match the snapshot template", () => {
        const {asFragment} = render(
            <SCThemeProvider theme={mockTheme}>
                <SocialMedia {...(props as any)} />
            </SCThemeProvider>,
        )

        expect(asFragment()).toMatchSnapshot()
    })

    describe(" - conditional elements", () => {
        it("should match the snapshot template when url is null in the elements", () => {
            const customiseData = {
                region: {
                    ...socialMediaData,
                    subRegions: [
                        {
                            title: null,
                            accessibilityTitle: null,
                            elements: [
                                {
                                    type: "Link",
                                    url: null,
                                    openInNewWindow: true,
                                    name: "Facebook",
                                    icon: "nx-icon facebook-black",
                                    text: null,
                                    accessibilityText: "Amido Official Facebook page (opens in a new window)",
                                    tooltip: null,
                                    accessibilityTooltip: null,
                                    description: null,
                                    accessibilityDescription: null,
                                },
                            ],
                        },
                    ],
                },
            }
            const {asFragment} = render(
                <SCThemeProvider theme={mockTheme}>
                    <SocialMedia {...(customiseData as any)} />
                </SCThemeProvider>,
            )

            expect(asFragment()).toMatchSnapshot()
        })
        it("should match the snapshot template when accessibilityText is null in the elements", () => {
            const customiseData = {
                region: {
                    ...socialMediaData,
                    subRegions: [
                        {
                            title: null,
                            accessibilityTitle: null,
                            elements: [
                                {
                                    type: "Link",
                                    url: "https://www.facebook.com/amidoofficial",
                                    openInNewWindow: true,
                                    name: "Facebook",
                                    icon: "nx-icon facebook-black",
                                    text: null,
                                    accessibilityText: null,
                                    tooltip: null,
                                    accessibilityTooltip: null,
                                    description: null,
                                    accessibilityDescription: null,
                                },
                            ],
                        },
                    ],
                },
            }
            const {asFragment} = render(
                <SCThemeProvider theme={mockTheme}>
                    <SocialMedia {...(customiseData as any)} />
                </SCThemeProvider>,
            )

            expect(asFragment()).toMatchSnapshot()
        })
        it("should match the snapshot template when url and accessibilityText is null in the elements", () => {
            const customiseData = {
                region: {
                    ...socialMediaData,
                    subRegions: [
                        {
                            title: null,
                            accessibilityTitle: null,
                            elements: [
                                {
                                    type: "Link",
                                    url: null,
                                    openInNewWindow: true,
                                    name: "Facebook",
                                    icon: "nx-icon facebook-black",
                                    text: null,
                                    accessibilityText: "",
                                    tooltip: null,
                                    accessibilityTooltip: null,
                                    description: null,
                                    accessibilityDescription: null,
                                },
                            ],
                        },
                    ],
                },
            }
            const {asFragment} = render(
                <SCThemeProvider theme={mockTheme}>
                    <SocialMedia {...(customiseData as any)} />
                </SCThemeProvider>,
            )

            expect(asFragment()).toMatchSnapshot()
        })
    })
})
