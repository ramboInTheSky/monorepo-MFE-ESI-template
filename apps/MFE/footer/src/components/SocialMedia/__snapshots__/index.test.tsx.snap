// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - SocialMedia:   - conditional elements should match the snapshot template when accessibilityText is null in the elements 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-center MuiGrid-grid-xs-12"
  >
    <h5
      class="MuiTypography-root components__SocialTitle-sc-1h8wyu0-0 eqGyQA MuiTypography-h5"
    >
      Our Social Networks
    </h5>
  </div>
</DocumentFragment>
`;

exports[`Components - SocialMedia:   - conditional elements should match the snapshot template when url and accessibilityText is null in the elements 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-center MuiGrid-grid-xs-12"
  >
    <h5
      class="MuiTypography-root components__SocialTitle-sc-1h8wyu0-0 eqGyQA MuiTypography-h5"
    >
      Our Social Networks
    </h5>
  </div>
</DocumentFragment>
`;

exports[`Components - SocialMedia:   - conditional elements should match the snapshot template when url is null in the elements 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-center MuiGrid-grid-xs-12"
  >
    <h5
      class="MuiTypography-root components__SocialTitle-sc-1h8wyu0-0 eqGyQA MuiTypography-h5"
    >
      Our Social Networks
    </h5>
  </div>
</DocumentFragment>
`;

exports[`Components - SocialMedia:  should match the snapshot template 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-center MuiGrid-grid-xs-12"
  >
    <h5
      class="MuiTypography-root components__SocialTitle-sc-1h8wyu0-0 eqGyQA MuiTypography-h5"
    >
      Our Social Networks
    </h5>
    <a
      aria-label="Amido Official Facebook page (opens in a new window)"
      class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      data-testid="footer-social-media-facebook"
      href="https://www.facebook.com/amidoofficial"
      target="_blank"
    >
      <img
        aria-hidden="true"
        class="components__SocialLinkIcon-sc-1h8wyu0-1 gLhcnT"
        src="/static-content/icons/footer/amido/default/icon-social-facebook.svg"
      />
    </a>
    <a
      aria-label="Amido Official Twitter page (opens in a new window)"
      class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      data-testid="footer-social-media-twitter"
      href="https://www.twitter.com/amidoofficial"
      target="_blank"
    >
      <img
        aria-hidden="true"
        class="components__SocialLinkIcon-sc-1h8wyu0-1 gLhcnT"
        src="/static-content/icons/footer/amido/default/icon-social-twitter.svg"
      />
    </a>
    <a
      aria-label="Amido Official Instagram page (opens in a new window)"
      class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      data-testid="footer-social-media-instagram"
      href="https://instagram.com/amidoofficial"
      target="_blank"
    >
      <img
        aria-hidden="true"
        class="components__SocialLinkIcon-sc-1h8wyu0-1 gLhcnT"
        src="/static-content/icons/footer/amido/default/icon-social-instagram.svg"
      />
    </a>
    <a
      aria-label="pinterest"
      class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      data-testid="footer-social-media-pinterest"
      href="pinterest.com/amidoofficial"
      target="_blank"
    >
      <img
        aria-hidden="true"
        class="components__SocialLinkIcon-sc-1h8wyu0-1 gLhcnT"
        src="/static-content/icons/footer/amido/default/icon-social-pinterest.svg"
      />
    </a>
    <a
      aria-label="Amido Shopping Youtube page (opens in a new window)"
      class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      data-testid="footer-social-media-youtube"
      href="https://www.youtube.com/user/amidoshopping"
      target="_blank"
    >
      <img
        aria-hidden="true"
        class="components__SocialLinkIcon-sc-1h8wyu0-1 gLhcnT"
        src="/static-content/icons/footer/amido/default/icon-social-youtube.svg"
      />
    </a>
  </div>
</DocumentFragment>
`;
